fpt/src/debugger.rs

```diff
@@ -0,0 +1,54 @@
+use crate::debug_interface::DebugCmd;
+
+#[derive(Clone, PartialEq)]
+pub struct Breakpoint {
+    pub pc: u16,
+    pub active: bool,
+}
+#[derive(Clone, PartialEq)]
+pub struct Watchpoint {
+    pub addr: u16,
+}
+
+#[derive(Clone, PartialEq)]
+pub struct Debugger {
+    breakpoints: Vec<Breakpoint>,
+    watchpoints: Vec<Watchpoint>,
+    pub paused: bool,
+}
+
+impl Debugger {
+    pub fn new() -> Self {
+        Self {
+            breakpoints: Vec::new(),
+            watchpoints: Vec::new(),
+            paused: false,
+        }
+    }
+
+    pub fn receive_command(&mut self, cmd: &DebugCmd) {
+        match cmd {
+            DebugCmd::Pause => {
+                self.paused = true;
+            }
+            DebugCmd::Continue => {
+                self.paused = false;
+            }
+            DebugCmd::Breakpoint(pc) => {
+                self.breakpoints.push(Breakpoint {
+                    pc: *pc,
+                    active: true,
+                });
```

Comment:
What if there's already a `Breakpoint` with the same `pc`?

- Maybe we should test for the existence of a breakpoint on this `pc` and do nothing in that case.
- Maybe the field `breakpoints` should be a [HashSet](https://doc.rust-lang.org/std/collections/hash_set/struct.HashSet.html) and `Breakpoint` should `impl PartialEq` to only compare the `pc` field.
- Maybe the field `breakpoints` should be a `HashMap<u16, bool>` that maps "program counters" to an active/disabled flag, or a simpler struct that for now only holds the active/disabled flag. The latter would allow us to extend the value later to hold, for example, a conditional expression. Or we could just make a separate `ConditionalBreakpoint` variant.